src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "NavigatorRenderer.hpp"
+#include "Formatter/Units.hpp"
+#include "InfoBoxes/InfoBoxLayout.hpp"
+#include "Interface.hpp"
+#include "Language/Language.hpp"
+#include "Look/FontDescription.hpp"
+#include "Look/Look.hpp"
+#include "Math/Angle.hpp"
+#include "NextArrowRenderer.hpp"
+#include "Renderer/TextRenderer.hpp"
+#include "Task/TaskBehaviour.hpp"
+#include "Units/System.hpp"
+#include "Waypoint/Waypoint.hpp"
+#include "WaypointIconRenderer.hpp"
+#include "WaypointRenderer.hpp"
+#include "time/RoughTime.hpp"
+#include "ui/canvas/Canvas.hpp"
+#include "Screen/Layout.hpp"
+#include "UIGlobals.hpp"
+#include "Look/TaskLook.hpp"
+#include "Look/WaypointLook.hpp"
+#include "Look/NavigatorLook.hpp"
+#include "Look/IconLook.hpp"
+#include "Look/MapLook.hpp"
+#include "Look/TaskLook.hpp"
+#include "Math/Constants.hpp"
+#include "ProgressBarRenderer.hpp"
+#include "Task/ProtectedTaskManager.hpp"
+#include "Engine/Task/TaskManager.hpp"
+#include "Engine/Task/Ordered/OrderedTask.hpp"
+#include "Engine/Task/Ordered/Points/OrderedTaskPoint.hpp"
+#include "Components.hpp"
+#include "Engine/Task/Unordered/AlternateList.hpp"
+#include "Formatter/LocalTimeFormatter.hpp"
+#include "Formatter/TimeFormatter.hpp"
+#include "ui/canvas/Font.hpp"
+#include "ui/dim/Rect.hpp"
+
+
+// standard
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+
+void
+NavigatorRenderer::DrawFrame(Canvas &canvas, const PixelRect &rc, const NavigatorLook &look) {
+  const auto top_left = rc.GetTopLeft();
+  const int rc_height = rc.GetHeight();
+  const int rc_width = rc.GetWidth();
+  
+  BulkPixelPoint polyline_frame[11];
+  polyline_frame[0].x = top_left.x + 5 / 20.0 * rc_height; //
```

Comment:
No floats, please! `5 / 20.0 * rc_height == rc_height / 4`, duh!

Reply:
Thanks for this tip (float to int)!
I deliberately didn't simplify the division to understand / remember how the frame was designed (fraction of rc_height). this could help in future to change the frame design.